src/multiparser.js

```diff
@@ -159,6 +163,62 @@ function fromHtmlParser2(path, options) {
   }
 }
 
+function wrapCssTemplate(quasisDoc, expressionDocs) {
+  const allReplaced = replacePlaceholders(quasisDoc, expressionDocs);
+  if (!allReplaced) {
+    throw new Error("Couldn't insert all the expressions");
+  }
+  return concat([
+    "`",
+    indent(concat([softline, stripTrailingHardline(quasisDoc)])),
+    softline,
+    "`"
+  ]);
+}
+
+// Search all the placeholders in the quasisDoc tree
+// and replace them with the expression docs one by one
+// returns true if all the expressions were replaced, false otherwise
+function replacePlaceholders(quasisDoc, expressionDocs) {
+  if (!expressionDocs || !expressionDocs.length) {
+    return true;
+  }
+  const expressions = expressionDocs.slice();
+  traverseDoc(quasisDoc, doc => {
+    if (!doc || !doc.parts || !doc.parts.length) {
+      // Recurse
+      return true;
+    }
+    if (
+      doc.parts.length > 1 &&
+      doc.parts[0] === "@" &&
+      typeof doc.parts[1] === "string" &&
+      doc.parts[1].startsWith("prettier-placeholder")
+    ) {
+      // If placeholder is splitted, join it
+      const [at, placeholder, ...rest] = doc.parts;
+      doc.parts = [at + placeholder, ...rest];
```

Comment:
I'm having problems trying to find a clean way to re-create the doc. Is the mutation during the iteration what you don't like? If so, I can postpone the mutations until the traverse finishes.